rasa/validator.py

```diff
@@ -483,3 +492,177 @@ def warn_if_config_mandatory_keys_are_not_set(self) -> None:
+                    f"id '{flow.id}'. Flow names must be unique. "
+                    f"Please make sure that all flows have different names."
+                )
+
+            flows_mapping[flow.name] = cleaned_description
+
+        return True
+
+    @staticmethod
+    def _construct_predicate(predicate: Optional[str], step_id: str) -> Predicate:
+        try:
+            pred = Predicate(predicate)
+        except (TypeError, Exception) as exception:
+            raise RasaException(
+                f"Could not initialize the predicate found under step "
+                f"'{step_id}'. Please make sure that all predicates "
+                f"are strings."
+            ) from exception
+
+        return pred
+
+    def verify_predicates(self) -> bool:
+        """Checks that predicates used in branch flow steps or `collect` steps are valid."""  # noqa: E501
+        all_good = True
+        for flow in self.flows.underlying_flows:
+            for step in flow.steps:
+                if isinstance(step, BranchFlowStep):
+                    for link in step.next.links:
+                        if isinstance(link, IfFlowLink):
+                            predicate = Validator._construct_predicate(
+                                link.condition, step.id
+                            )
+                            if not predicate.is_valid():
+                                raise RasaException(
+                                    f"Detected invalid condition '{link.condition}' "
+                                    f"at step '{step.id}' for flow id '{flow.id}'. "
+                                    f"Please make sure that all conditions are valid."
+                                )
+                elif isinstance(step, CollectInformationFlowStep):
+                    predicates = [predicate.if_ for predicate in step.rejections]
+                    for predicate in predicates:
+                        pred = Validator._construct_predicate(predicate, step.id)
+                        if not pred.is_valid():
+                            raise RasaException(
+                                f"Detected invalid rejection '{predicate}' "
+                                f"at `collect` step '{step.id}' "
+                                f"for flow id '{flow.id}'. "
+                                f"Please make sure that all conditions are valid."
+                            )
+        return all_good
+
+    def verify_flows(self) -> bool:
+        """Checks for inconsistencies across flows."""
+        logger.info("Validating flows...")
+
+        if self.flows.is_empty():
+            logger.warning(
+                "No flows were found in the data files. "
+                "Will not proceed with flow validation.",
+            )
+            return True
+
+        all_good = (
+            self.verify_flows_steps_against_domain()
+            and self.verify_unique_flows()
+            and self.verify_predicates()
+        )
```

Comment:
Why use this if it always raises the exception (and stops) in case of a validation fail?

Reply:
I was just trying to find a bridge with the existing approach in `cli/utils.py` that expects a bool if everything went fine.